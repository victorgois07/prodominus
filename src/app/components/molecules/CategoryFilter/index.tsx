import { Label } from "../../atoms";

interface CategoryFilterProps {
  categories: string[];
  selected: string;
  onSelect: (category: string) => void;
}

export function CategoryFilter({
  categories,
  selected,
  onSelect,
}: CategoryFilterProps) {
  return (
    <div className="mb-6">
      <Label htmlFor="category-filter">Categoria</Label>
      <select
        id="category-filter"
        className="block w-full p-2 border border-gray-300 rounded"
        value={selected}
        onChange={(e) => onSelect(e.target.value)}
      >
        <option value="">Todas</option>
        {categories.map((cat) => (
          <option key={cat} value={cat}>
            {cat}
          </option>
        ))}
      </select>
    </div>
  );
}
