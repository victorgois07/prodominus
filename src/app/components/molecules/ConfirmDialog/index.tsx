import { useConfirmDialog } from "@/business/store/useConfirmDialog";

export function ConfirmDialog() {
  const { open, message, onConfirm, onCancel, close } = useConfirmDialog();

  if (!open) return null;

  return (
    <div className="fixed inset-0 z-50 flex items-center justify-center bg-black/40">
      <div className="bg-white rounded-lg shadow-lg p-6 w-full max-w-sm">
        <div className="mb-4 text-gray-800">{message}</div>
        <div className="flex justify-end gap-2">
          <button
            className="px-4 py-2 rounded bg-gray-200 hover:bg-gray-300"
            onClick={() => {
              close();
              onCancel?.();
            }}
          >
            Cancelar
          </button>
          <button
            className="px-4 py-2 rounded bg-red-600 text-white hover:bg-red-700"
            onClick={() => {
              close();
              onConfirm?.();
            }}
          >
            Confirmar
          </button>
        </div>
      </div>
    </div>
  );
}
