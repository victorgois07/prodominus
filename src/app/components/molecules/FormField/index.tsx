import { ReactNode } from "react";
import { Label } from "../../atoms";
interface FormFieldProps {
  label: string;
  htmlFor: string;
  error?: string;
  children: ReactNode;
}

export function FormField({ label, htmlFor, error, children }: FormFieldProps) {
  return (
    <div>
      <Label htmlFor={htmlFor}>{label}</Label>
      {children}
      {error && <span className="text-red-500 text-sm">{error}</span>}
    </div>
  );
}
