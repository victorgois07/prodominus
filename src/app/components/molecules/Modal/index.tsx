import { useEffect, useRef } from "react";

interface ModalProps {
  open: boolean;
  onClose: () => void;
  children: React.ReactNode;
  title?: string;
  size?: "sm" | "md" | "lg";
  position?: "top" | "center";
}

export function Modal({
  open,
  onClose,
  children,
  title,
  size = "md",
  position = "center",
}: ModalProps) {
  const modalRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    const handleEscape = (e: KeyboardEvent) => {
      if (e.key === "Escape") onClose();
    };

    const handleClickOutside = (e: MouseEvent) => {
      if (modalRef.current && !modalRef.current.contains(e.target as Node)) {
        onClose();
      }
    };

    if (open) {
      document.addEventListener("keydown", handleEscape);
      document.addEventListener("mousedown", handleClickOutside);
      document.body.style.overflow = "hidden";
    }

    return () => {
      document.removeEventListener("keydown", handleEscape);
      document.removeEventListener("mousedown", handleClickOutside);
      document.body.style.overflow = "unset";
    };
  }, [open, onClose]);

  if (!open) return null;

  const sizeClasses = {
    sm: "max-w-sm",
    md: "max-w-md",
    lg: "max-w-2xl",
  };

  const positionClasses = {
    top: "mt-20",
    center: "mt-0",
  };

  return (
    <div
      data-testid="modal-overlay"
      className="fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center z-50"
    >
      <div
        ref={modalRef}
        role="dialog"
        className={`bg-white rounded-lg p-6 ${sizeClasses[size]} w-full mx-4 max-h-[90vh] overflow-y-auto ${positionClasses[position]}`}
      >
        {title && (
          <div className="flex justify-between items-center mb-4">
            <h2 className="text-xl font-bold">{title}</h2>
            <button
              onClick={onClose}
              className="text-gray-500 hover:text-gray-700"
              aria-label="close"
            >
              âœ•
            </button>
          </div>
        )}
        {children}
      </div>
    </div>
  );
}
