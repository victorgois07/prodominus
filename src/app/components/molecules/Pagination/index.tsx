import { Button } from "../../atoms";

interface PaginationProps {
  page: number;
  totalPages: number;
  onPage: (page: number) => void;
}

export function Pagination({ page, totalPages, onPage }: PaginationProps) {
  if (totalPages <= 1) return null;

  return (
    <nav className="flex justify-center my-8">
      <ul className="inline-flex items-center gap-1">
        <li>
          <Button onClick={() => onPage(page - 1)} disabled={page === 1}>
            Anterior
          </Button>
        </li>
        {Array.from({ length: totalPages }, (_, i) => i + 1).map((p) => (
          <li key={p}>
            <Button
              onClick={() => onPage(p)}
              disabled={p === page}
              className={
                p === page
                  ? "bg-green-500 text-white border border-green-600"
                  : "bg-white border border-gray-300"
              }
            >
              {p}
            </Button>
          </li>
        ))}
        <li>
          <Button
            onClick={() => onPage(page + 1)}
            disabled={page === totalPages}
          >
            Pr√≥xima
          </Button>
        </li>
      </ul>
    </nav>
  );
}
