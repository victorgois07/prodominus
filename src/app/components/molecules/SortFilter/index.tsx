import { Label } from "../../atoms";

interface SortFilterProps {
  sort: string;
  onSort: (sort: string) => void;
}

export function SortFilter({ sort, onSort }: SortFilterProps) {
  return (
    <div className="mb-6">
      <Label htmlFor="sort-filter">Ordenar por</Label>
      <select
        id="sort-filter"
        className="block w-full p-2 border border-gray-300 rounded"
        value={sort}
        onChange={(e) => onSort(e.target.value)}
      >
        <option value="">Padrão</option>
        <option value="asc">Preço: Menor para Maior</option>
        <option value="desc">Preço: Maior para Menor</option>
      </select>
    </div>
  );
}
