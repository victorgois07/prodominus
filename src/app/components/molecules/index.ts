export * from "./CategoryFilter";
export * from "./ConfirmDialog";
export * from "./FormField";
export * from "./Modal";
export * from "./Pagination";
export * from "./SortFilter";
