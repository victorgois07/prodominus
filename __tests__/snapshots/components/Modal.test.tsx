import { Modal } from "@/app/components/molecules/Modal";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Modal Snapshot", () => {
  it("should match snapshot with title and children", () => {
    const { container } = render(
      <Modal title="Test Modal" isOpen={true} onClose={() => {}}>
        <div>Modal Content</div>
      </Modal>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with long title", () => {
    const { container } = render(
      <Modal
        title="This is a very long modal title that should be properly displayed"
        isOpen={true}
        onClose={() => {}}
      >
        <div>Modal Content</div>
      </Modal>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with complex content", () => {
    const { container } = render(
      <Modal title="Complex Modal" isOpen={true} onClose={() => {}}>
        <div>
          <h2>Section 1</h2>
          <p>This is a paragraph with some text.</p>
          <ul>
            <li>Item 1</li>
            <li>Item 2</li>
            <li>Item 3</li>
          </ul>
          <h2>Section 2</h2>
          <p>Another paragraph with more text.</p>
          <button>Click me</button>
        </div>
      </Modal>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot when closed", () => {
    const { container } = render(
      <Modal title="Closed Modal" isOpen={false} onClose={() => {}}>
        <div>Modal Content</div>
      </Modal>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with custom size", () => {
    const { container } = render(
      <Modal
        title="Custom Size Modal"
        isOpen={true}
        onClose={() => {}}
        size="lg"
      >
        <div>Modal Content</div>
      </Modal>
    );

    expect(container).toMatchSnapshot();
  });
});
