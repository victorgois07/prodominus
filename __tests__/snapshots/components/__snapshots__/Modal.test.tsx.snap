// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal Snapshot > should match snapshot when closed 1`] = `<div />`;

exports[`Modal Snapshot > should match snapshot with complex content 1`] = `<div />`;

exports[`Modal Snapshot > should match snapshot with custom size 1`] = `<div />`;

exports[`Modal Snapshot > should match snapshot with long title 1`] = `<div />`;

exports[`Modal Snapshot > should match snapshot with title and children 1`] = `<div />`;
